Options for CachedMerklePatriciaTree

SoujanyaPonnapalli opened the pull request:
This PR adds support for MerklePatriciaTreeOptions to the CachedMerklePatriciaTree.
Resolves Issue #79

no2chem commented on src/index.ts:
```diff
@@ -1535,8 +1535,10 @@ export class CachedMerklePatriciaTree<K, V> extends MerklePatriciaTree<K, V> {
-    super();
+  constructor(
+      options: MerklePatriciaTreeOptions<K, V> = {putCanDelete: true},
+      depth = 6) {
```
You should use a public auto-property here:
```
constructor(options, public maxCacheDepth = 6);
```
Is there some reason maxCacheDepth is private anyway?

SoujanyaPonnapalli commented:
maxCacheDepth was private to avoid changing the depth of the cachedTree after initializing the cache. However, it makes sense to have the flexibility to change the max depth of the cache.

Addresses the comments and closes #82 
@no2chem please merge the PR if everything looks good.

SoujanyaPonnapalli commented:
closes #82
